internal/venice-client-common/src/main/java/com/linkedin/venice/D2/D2ClientUtils.java

```diff
@@ -112,7 +114,11 @@ public static RestResponse sendD2GetRequest(String requestPath, D2Client client)
     try {
       response = client.restRequest(createD2GetRequest(requestPath)).get();
     } catch (Exception e) {
-      throw new VeniceException("D2 client failed to sent request, " + requestPath, e);
+      Throwable t = ExceptionUtils.getRecursiveCause(e, RestException.class);
+      if (t == null) {
+        throw new VeniceException("D2 client failed to send request, " + requestPath, e);
+      }
+      response = ((RestException) t).getResponse();
```

Comment:
Why do we need this change? Is it ok not throwing exception when there is no RestException?